environment/src/main/java/com/sequenceiq/environment/environment/repository/EnvironmentRepository.java

```diff
@@ -45,18 +45,34 @@
             + "AND e.archived = false")
     Set<Environment> findAllByIdNotArchived(@Param("ids") List<Long> ids);
 
-    Set<Environment> findByNameInAndAccountIdAndArchivedIsFalse(Collection<String> names, String accountId);
-
-    Set<Environment> findByResourceCrnInAndAccountIdAndArchivedIsFalse(Collection<String> resourceCrns, String accountId);
-
-    Optional<Environment> findByNameAndAccountIdAndArchivedIsFalse(@Param("name") String name, @Param("accountId") String accountId);
-
     @Query("SELECT e.resourceCrn FROM Environment e " +
-            "WHERE e.accountId = :accountId " +
-            "AND e.name IN (:names) " +
+            "WHERE e.name in (:names) " +
+            "AND e.accountId = :accountId " +
             "AND e.archived = false")
-    List<String> findAllCrnByNameAndAccountIdAndArchivedIsFalse(@Param("names") Collection<String> names, @Param("accountId") String accountId);
+    List<String> findAllCrnByNameInAndAccountIdAndArchivedIsFalse(@Param("names") Collection<String> names, @Param("accountId") String accountId);
+
+    @Query("SELECT e FROM Environment e "
+            + "LEFT JOIN FETCH e.network n "
+            + "LEFT JOIN FETCH n.environment ev "
+            + "LEFT JOIN FETCH e.credential c "
+            + "LEFT JOIN FETCH e.authentication a "
+            + "LEFT JOIN FETCH e.parameters p "
+            + "WHERE e.accountId = :accountId "
+            + "AND e.name = :name "
+            + "AND e.archived = false")
+    Optional<Environment> findByNameAndAccountIdAndArchivedIsFalse(@Param("name") String name, @Param("accountId") String accountId);
 
+    @Query("SELECT e FROM Environment e "
+            + "LEFT JOIN FETCH e.network n "
+            + "LEFT JOIN FETCH n.environment ev "
+            + "LEFT JOIN FETCH e.credential c "
+            + "LEFT JOIN FETCH e.authentication a "
+            + "LEFT JOIN FETCH e.parameters p "
+            + "LEFT JOIN FETCH e.proxyConfig p "
```

Comment:
Please correct the fetched references as `p` used multiple times and could result issues in JPA when the query is used.